Logout button sometimes doesn't log out

The exception that occurs is caught but not informing the user about the supposed logout failure seems like bad UX.

12-17 13:28:46.292    5033-5052/org.apereo E/org.apereo.activities.HomePageActivity﹕ null
    org.springframework.web.client.ResourceAccessException: I/O error: Unexpected status line: PNG; nested exception is java.net.ProtocolException: Unexpected status line: PNG
            at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:491)
            at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:439)
            at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:415)
            at org.apereo.services.RestInterface_.getMainFeed(RestInterface_.java:116)
            at org.apereo.services.RestApi.getMainFeed(RestApi.java:83)
            at org.apereo.services.RestApi_.access$001(RestApi_.java:12)
            at org.apereo.services.RestApi_$1.execute(RestApi_.java:48)
            at org.androidannotations.api.BackgroundExecutor$Task.run(BackgroundExecutor.java:393)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.net.ProtocolException: Unexpected status line: PNG
            at com.android.okhttp.internal.http.RawHeaders.setStatusLine(RawHeaders.java:108)
            at com.android.okhttp.internal.http.RawHeaders.fromBytes(RawHeaders.java:308)
            at com.android.okhttp.internal.http.HttpTransport.readResponseHeaders(HttpTransport.java:135)
            at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:644)
            at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:347)
            at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:296)
            at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:503)
            at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:136)
            at org.springframework.http.client.SimpleClientHttpResponse.getRawStatusCode(SimpleClientHttpResponse.java:54)
            at org.springframework.http.client.SimpleClientHttpResponse.getStatusCode(SimpleClientHttpResponse.java:80)
            at org.springframework.web.client.DefaultResponseErrorHandler.hasError(DefaultResponseErrorHandler.java:46)
            at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:477)
            at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:439)
            at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:415)
            at org.apereo.services.RestInterface_.getMainFeed(RestInterface_.java:116)
            at org.apereo.services.RestApi.getMainFeed(RestApi.java:83)
            at org.apereo.services.RestApi_.access$001(RestApi_.java:12)
            at org.apereo.services.RestApi_$1.execute(RestApi_.java:48)
            at org.androidannotations.api.BackgroundExecutor$Task.run(BackgroundExecutor.java:393)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)

--- a/app/src/main/java/org/apereo/activities/HomePageActivity.java
+++ b/app/src/main/java/org/apereo/activities/HomePageActivity.java
@@ -175,17 +175,13 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
             @Override
             public void onError(Exception e, Integer response) {
                 Logger.e(TAG, "error logging out (received status code " + response + ")", e);
+                dismissSpinner();
+                showLongToast(getString(R.string.error_logging_out));
             }
 
             @Override
             public void onSuccess(Integer response) {
                 getFeed();
-            }
-
-            @Override
-            public void onFinish() {
-                super.onFinish();
-                dismissSpinner();
             }
         });
     }
@@ -242,6 +238,7 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
             @Override
             public void onError(Exception e, String responseBody) {
                 Logger.e(TAG, responseBody, e);
+                dismissSpinner();
             }
 
             @Override
@@ -254,12 +251,6 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
                 layoutManager.setLayout(layout);
 
                 reconfigureViews();
-            }
-
-            @Override
-            public void onFinish() {
-                super.onFinish();
-                dismissSpinner();
             }
         });
     }
@@ -290,6 +281,8 @@ public class HomePageActivity extends BaseActivity implements IActionListener, A
 
         // Switch back to the first tab.
         selectItem(0);
+
+        dismissSpinner();
     }
 
     @UiThread

